src/lib.rs

```diff
@@ -812,447 +825,453 @@ impl Default for Section {
+            })
             .get_contents()
             .unwrap(),
-        &[0xab, 1, 0xcd, 0xef]
-    );
-}
+            &[0xab, 4, 0xcd, 0xef]
+        );
+    }
 
-#[test]
-fn section_append_repeated() {
-    let s = Section::new();
-    assert_eq!(
-        s.append_repeated(0xff, 5).get_contents().unwrap(),
-        &[0xff, 0xff, 0xff, 0xff, 0xff]
-    );
-}
+    #[test]
+    fn section_append_repeated() {
+        assert_eq!(
+            Section::inline(None, |s| s.append_repeated(0xff, 5))
+                .get_contents()
+                .unwrap(),
+            &[0xff, 0xff, 0xff, 0xff, 0xff]
+        );
+    }
 
-#[test]
-fn section_align() {
-    let s = Section::new();
-    assert_eq!(
-        s.D8(1).align(8).D8(1).get_contents().unwrap(),
-        &[1, 0, 0, 0, 0, 0, 0, 0, 1]
-    );
-}
+    #[test]
+    fn section_align() {
+        assert_eq!(
+            Section::inline(None, |s| s.D8(1).align(8).D8(1))
+                .get_contents()
+                .unwrap(),
+            &[1, 0, 0, 0, 0, 0, 0, 0, 1]
+        );
+    }
 
-#[test]
-fn section_test_8() {
-    let s = Section::new();
-    assert_eq!(
-        s.D8(0x12).L8(0x12).B8(0x12).get_contents().unwrap(),
-        &[0x12, 0x12, 0x12]
-    );
-}
+    #[test]
+    fn section_test_8() {
+        assert_eq!(
+            Section::inline(None, |s| s.D8(0x12).L8(0x12).B8(0x12))
+                .get_contents()
+                .unwrap(),
+            &[0x12, 0x12, 0x12]
+        );
+    }
 
-#[test]
-fn section_test_16() {
-    let s = Section::with_endian(Endian::Little);
-    assert_eq!(
-        s.D16(0xABCD)
-            .L16(0xABCD)
-            .B16(0xABCD)
+    #[test]
+    fn section_test_16() {
+        assert_eq!(
+            Section::inline(Some(Endian::Little), |s| s
+                .D16(0xABCD)
+                .L16(0xABCD)
+                .B16(0xABCD))
             .get_contents()
             .unwrap(),
-        &[0xCD, 0xAB, 0xCD, 0xAB, 0xAB, 0xCD]
-    );
-}
+            &[0xCD, 0xAB, 0xCD, 0xAB, 0xAB, 0xCD]
+        );
+    }
 
-#[test]
-fn section_test_32() {
-    let s = Section::with_endian(Endian::Little);
-    assert_eq!(
-        s.D32(0xABCD1234)
-            .L32(0xABCD1234)
-            .B32(0xABCD1234)
+    #[test]
+    fn section_test_32() {
+        assert_eq!(
+            Section::inline(Some(Endian::Little), |s| s
+                .D32(0xABCD1234)
+                .L32(0xABCD1234)
+                .B32(0xABCD1234))
             .get_contents()
             .unwrap(),
-        &[0x34, 0x12, 0xCD, 0xAB, 0x34, 0x12, 0xCD, 0xAB, 0xAB, 0xCD, 0x12, 0x34]
-    );
-}
+            &[0x34, 0x12, 0xCD, 0xAB, 0x34, 0x12, 0xCD, 0xAB, 0xAB, 0xCD, 0x12, 0x34]
+        );
+    }
 
-#[test]
-fn section_test_64() {
-    let s = Section::with_endian(Endian::Little);
-    assert_eq!(
-        s.D64(0x12345678ABCDEFFF)
-            .L64(0x12345678ABCDEFFF)
-            .B64(0x12345678ABCDEFFF)
+    #[test]
+    fn section_test_64() {
```

Comment:
I should probably look into codegenning these tests, or using a property-based testing framework, or something like that.

Reply:
Heh, yah, some of these could be made a bit smaller. I've been using https://docs.rs/rstest/0.11.0/rstest/ which is quite nice since it is similar to python's `pytest`.